goalie/go_mesh_seq.py

```diff
@@ -228,6 +234,31 @@ def indicate_errors(
                     indi.interpolate(abs(indi))
                     self.indicators[f][i][j].interpolate(ufl.max_value(indi, 1.0e-16))
 
+            # discard current subinterval duplicate solution fields
+            if not self.steady:
+                for f in self.fields:
+                    self.solutions[f][FWD_OLD].pop(-1)
+                    self.solutions[f][ADJ_NEXT].pop(-1)
+                    enriched_mesh_seq.solutions[f][FWD_OLD].pop(-1)
+                    enriched_mesh_seq.solutions[f][ADJ_NEXT].pop(-1)
+
+            # delete current subinterval enriched mesh to reduce the memory footprint
+            if len(enriched_mesh_seq.meshes) > 1:
+                for f in self.fields:
+                    enriched_mesh_seq._fs[f].pop(-1)
+                enriched_mesh_seq.meshes.pop(-1)
+                enriched_mesh_seq.time_partition.drop_last_subinterval()
+
+        # clear empty labels
+        for f in self.fields:
+            if self.steady:
+                self.solutions.labels = ("forward",)
+                self.solutions[f].pop("forward_old", None)
+            else:
+                self.solutions.labels = ("forward", "adjoint")
+                self.solutions[f].pop("forward_old", None)
+                self.solutions[f].pop("adjoint_next", None)
+
```

Comment:
`indicate_errors` is already quite a bloated method so it'd be nice if these were put in separate method(s), similarly to the one you introduced for `TimePartition`.